features/events/traps/src/main/java/org/opennms/netmgt/trapd/TrapListener.java

```diff
@@ -56,6 +59,9 @@
 public class TrapListener implements TrapNotificationListener {
     private static final Logger LOG = LoggerFactory.getLogger(TrapListener.class);
 
+    // True if trap configuration has been received from the core instance
+    private AtomicBoolean m_configured = new AtomicBoolean(false);
```

Comment:
Can't we use `m_registeredForTraps` to make the decision to open or not ?

Reply:
`m_registeredForTraps` is updated near the middle of the `open` method. There's a chance that the TimerTask could wake up and see `m_registeredForTraps=false` and attempt to use default settings while an update is already in progress. i didn't want to move where the value is set because it's in the correct spot, so I just made another variable.